service/src/main/java/tds/assessment/repositories/impl/AssessmentQueryRepositoryImpl.java

```diff
@@ -55,9 +59,38 @@ public AssessmentQueryRepositoryImpl(final DataSource dataSource) {
         if (rows.isEmpty()) {
             logger.debug("Did not findAssessmentByKey a result for assessment from tblsetofadminsubjects for %s", assessmentKey);
         } else {
-            maybeAssessment = assessmentMapper.mapResults(rows);
+            List<Form> forms = findFormsForAssessment(parameters);
+            maybeAssessment = assessmentMapper.mapResults(rows, forms);
         }
 
         return maybeAssessment;
     }
+
+    private List<Form> findFormsForAssessment(SqlParameterSource parameters) {
+        final String formsSQL =
+                "SELECT \n" +
+                        "    segments._key AS segmentKey, \n" +
+                        "    forms._key AS `key`, \n" +
+                        "    forms.formid AS id, \n" +
+                        "    forms.language, \n" +
+                        "    forms.loadconfig AS loadVersion, \n" +
+                        "    forms.updateconfig AS updateVersion, \n" +
+                        "    forms.cohort \n" +
+                        "FROM itembank.tblsetofadminsubjects segments \n" +
```

Comment:
Since we're specifically trying to get forms here I think this should go against itembank.testform and join on segments.  No left join.

`FROM itembank.testform forms JOIN itembank.tblsetofadminsubjects segements on segments._key = forms._fk_adminsubject`

I don't see why we'd want to get records back if the join isn't successful which the left will do.